src/Raven.Server/Documents/Handlers/Processors/Indexes/IndexHandlerProcessorForConvertAutoIndex.cs

```diff
@@ -0,0 +1,716 @@
+                .AppendLine("select new")
+                .AppendLine("{");
+
+            HandleMapFields(sb, autoIndex, context);
+
+            sb.AppendLine("};");
+
+            return [sb.ToString()];
+        }
+
+        static string ConstructReduce(AutoIndexDefinition autoIndex)
+        {
+            if (autoIndex.GroupByFields == null || autoIndex.GroupByFields.Count == 0)
+                return null;
+
+            var sb = new StringBuilder();
+
+            ConstructReduceInternal(sb, autoIndex);
+
+            return sb.ToString();
+        }
+
+        static Dictionary<string, IndexFieldOptions> ConstructFields(AutoIndexDefinition autoIndex, AutoIndexConversionContext context)
+        {
+            Dictionary<string, IndexFieldOptions> fields = null;
+            if (autoIndex.MapFields is { Count: > 0 })
+            {
+                foreach (var kvp in autoIndex.MapFields)
+                {
+                    var fieldNames = GenerateFieldName(kvp.Key, kvp.Value.Indexing);
+
+                    foreach (var f in fieldNames)
+                    {
+                        HandleFieldIndexing(f.FieldName, f.Indexing);
+                        HandleSpatial(f.FieldName, kvp.Value.Spatial, context);
+                        HandleStorage(f.FieldName, kvp.Value.Storage);
+                        HandleSuggestions(f.FieldName, kvp.Value.Suggestions);
+                    }
+                }
+            }
+
+            return fields;
+
+            IndexFieldOptions GetFieldOptions(string fieldName)
+            {
+                fields ??= new Dictionary<string, IndexFieldOptions>();
+                if (fields.TryGetValue(fieldName, out var value) == false)
+                    fields[fieldName] = value = new IndexFieldOptions();
+
+                return value;
+            }
+
+            void HandleFieldIndexing(string fieldName, AutoFieldIndexing? fieldIndexing)
+            {
+                if (fieldIndexing.HasValue == false)
+                    return;
+
+                if (fieldIndexing.Value.HasFlag(AutoFieldIndexing.Search))
+                {
+                    var options = GetFieldOptions(fieldName);
+                    options.Indexing = FieldIndexing.Search;
+                    return;
+                }
+
+                if (fieldIndexing.Value.HasFlag(AutoFieldIndexing.Exact))
+                {
+                    var options = GetFieldOptions(fieldName);
+                    options.Indexing = FieldIndexing.Exact;
+                    return;
+                }
+
+                if (fieldIndexing.Value.HasFlag(AutoFieldIndexing.Highlighting))
+                {
+                    var options = GetFieldOptions(fieldName);
+                    options.Indexing = FieldIndexing.Search;
+                    options.Storage = FieldStorage.Yes;
+                    options.TermVector = FieldTermVector.WithPositionsAndOffsets;
+                    return;
+                }
```

Comment:
What about `AutoFieldIndexing.Default`? And what if we have multiple flags set?

Reply:
since this is a default then there is nothing to set